### PR TITLE
 Bump the phpstan analysis level to 4 

### PR DESCRIPTION
**Pull request by Jalle19**
@crisu83 this actually fixes some bugs so in my opinion it's important that we get it in.

### CODE REVIEW

File: src/Type/Coercer/IntCoercer.php [hunk 1]
```diff
@@ -19,13 +21,13 @@ public function coerce($value)
...
 
-        if (!\is_int($value) || $value > PHP_INT_MAX || $value < PHP_INT_MIN) {
+        if (!\is_numeric($value) || $value > PHP_INT_MAX || $value < PHP_INT_MIN) {
             throw new InvalidTypeException(
                 \sprintf('Int cannot represent non 32-bit signed integer value: %s', $value)
             );
         }
 
-        $intValue   = (int)$value;
+        $intValue   = $value;
```
Comment (spawnia) [inline src/Type/Coercer/IntCoercer.php]: That assignment is no longer needed, right?

Comment (Jalle19) [reply src/Type/Coercer/IntCoercer.php]: You're probably right

File: tests/Unit/Util/ValueConverterTest.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+     * @throws \Digia\GraphQL\Error\ConversionException
+     * @throws \Digia\GraphQL\Error\InvariantException
+     * @throws \Digia\GraphQL\Error\SyntaxErrorException
+     */
+    public function testIdType(): void
+    {
+        $idType = ID();
+
+        // Ensure numerical IDs become Ints
+        $this->assertInstanceOf(IntValueNode::class, $this->valueConverter->convert('45', $idType));
```
Comment (crisu83) [inline tests/Unit/Util/ValueConverterTest.php]: I would just call `ID()` multiple times for less code.

Comment (Jalle19) [reply tests/Unit/Util/ValueConverterTest.php]: Fixed

**Comment by Jalle19** [comment]
Scrutinizer complains about ValueConverter, `The condition $type === ID() is always false. `.

**Comment by Jalle19** [comment]
Maybe we should remove the special case handling for the ID type and just let it be a string?

**Comment by Jalle19** [comment]
I made a separate issue for the ID type issues so we can get this merged: https://github.com/digiaonline/graphql-php/issues/255

**Pull request merged**